src/AddressRegistry/StreetName/StreetNameAddress.cs

```diff
@@ -92,6 +92,21 @@ private void RejectBecauseParentWasRejected()
             }
         }
 
+        public void RejectBecauseStreetNameWasRetired()
+        {
+            if (IsRemoved)
+            {
+                return;
+            }
+
+            if (Status == AddressStatus.Rejected)
+            {
+                return;
+            }
```

Comment:
Don't we need a check to see if the address is proposed?

Reply:
We might have to turn around the if check here. `if(Status == Proposed) Apply(...)` else nothing happens.